homeassistant/components/frontend/__init__.py

```diff
@@ -345,14 +366,39 @@ def update_theme_and_fire_event():
             )
         hass.bus.async_fire(EVENT_THEMES_UPDATED)
 
-    @callback
-    def set_theme(call):
+    async def save_theme_store():
+        """Update store with new theme data."""
+        store = hass.data[DATA_THEMES_STORE]
+        data = {
+            DATA_DEFAULT_THEME: hass.data[DATA_DEFAULT_THEME],
+            DATA_DEFAULT_DARK_THEME: hass.data.get(DATA_DEFAULT_DARK_THEME),
+        }
+        await store.async_save(data)
```

Comment:
Let's do delay save in case we get two service calls to change both normal and dark theme.